File: python/src/services/GreetingService.py
from typing import List

from src.entities import Greeting
from src.repositories import GreetingRepository

class GreetingService:
    def __init__(self, greetingRepository: GreetingRepository) -> None:
        self._greetingRepository = greetingRepository
    
    def create(self, message: str) -> Greeting:
        g = Greeting(message)
        return self._greetingRepository.save(g)

    def getAllGreetings(self) -> List[Greeting]:
        return self._greetingRepository.findAll()

    def getGreeting(self, id: int) -> Greeting:
        g = self._greetingRepository.findById(id)
        if g is None:
            raise Exception("Greeting with id: " + str(id) + " not found!")
        return g

File: java/src/main/java/com/example/demo/services/GreetingService.java
package com.example.demo.services;

import java.util.List;

import com.example.demo.entities.Greeting;
import com.example.demo.repositories.IGreetingRepository;

public class GreetingService{

    private final IGreetingRepository greetingRepository;

    public GreetingService(IGreetingRepository greetingRepository) {
        this.greetingRepository = greetingRepository;
    }
    
    public Greeting create(String message){
        Greeting g = new Greeting(message);
        return greetingRepository.save(g);
    }

    public List<Greeting> getAllGreetings(){
        return greetingRepository.findAll();
    }

    public Greeting getGreeting(Long id) {
        return greetingRepository.findById(id).orElseThrow(() -> new RuntimeException("Greeting with id: " + id + " not found!"));
    }

}
